packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -1387,14 +1388,16 @@ class _CupertinoTextFieldState extends State<CupertinoTextField>
             // In the middle part, stack the placeholder on top of the main EditableText
             // if needed.
             Expanded(
-              child: Stack(
-                // Ideally this should be baseline aligned. However that comes at
-                // the cost of the ability to compute the intrinsic dimensions of
-                // this widget.
-                // See also https://github.com/flutter/flutter/issues/13715.
-                alignment: AlignmentDirectional.center,
-                textDirection: widget.textDirection,
-                children: <Widget>[if (placeholder != null) placeholder, editableText],
+              child: Directionality(
+                textDirection: widget.textDirection ?? Directionality.of(context),
+                child: _BaselineAlignedStack(
+                  placeholder: placeholder,
+                  editableText: editableText,
+                  textBaseline:
```

Comment:
Thanks for finding these on TextStyle. But I think you should use `textStyle`'s baseline type for editable text only and `placeholderStyle`'s for the placeholder.